### PR TITLE
fix tolerations types to always be string

### PR DESCRIPTION
**Pull request by eemurphy**
Signed-off-by: Erin Murphy <erinmurp@redhat.com>

issue: https://github.com/stolostron/backlog/issues/24856

### CODE REVIEW

File: pkg/rendering/renderer.go [hunk 1]
```diff
@@ -28,25 +30,120 @@ const (
...
+
+	v := reflect.ValueOf(u)
+	values := make([]string, reflect.Indirect(v).NumField())
+	var operator corev1.TolerationOperator = u.Operator
+	var effect corev1.TaintEffect = u.Effect
+
+	for i := 0; i < reflect.Indirect(v).NumField(); i++ {
+		fmt.Println(reflect.Indirect(v).Field(i).Kind())
+		fmt.Println(reflect.Indirect(v).Field(i).Type())
+		fmt.Println(reflect.Indirect(v).Field(i).Interface())
```
Comment (JakobGray) [inline pkg/rendering/renderer.go:86]: All the print statements used for debugging should come out

File: pkg/rendering/renderer.go [hunk 2]
```diff
@@ -28,25 +30,120 @@ const (
...
+		Operator:          operator,
+		Value:             values[2],
+		Effect:            effect,
+		TolerationSeconds: u.TolerationSeconds,
+	})
+}
+
+func (val *Values) ToValues() chartutil.Values {
+	inrec, _ := json.Marshal(val)
+	vals, _ := chartutil.ReadValues(inrec)
```
Comment (JakobGray) [inline pkg/rendering/renderer.go:145]: The errors here should not be ignored

**Review by JakobGray — CHANGES_REQUESTED** [review]

**Comment by ray-harris** [comment]
Is there any reason we're defining our own Toleration struct instead of using the k8s defined toleration struct?

https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3021-L3046

**Comment by eemurphy** [comment]
@ray-harris I may be incorrect, but we can not use the MarshalJSON() method with a type not defined inside the current package. I get this error if I try: `invalid receiver type "k8s.io/api/core/v1".Toleration (type not defined in this package)` 

I inherited some of this code from Jakob, and he made that change to just make our own version of Toleration so we could use the marshal method. I couldn't find any work around, though looking now maybe we could have just wrapped the k8s Toleration struct inside our own struct, which isn't too different from the change here

**Comment by JakobGray** [comment]
The reason the toleration struct was in my code was so I could play around with marshaling options without updating all the chart templates, because they're defined in the templates as `{{ if .Key }} key: {{ .Key }} {{- end }}` so the modification my struct had was to add that capitalization (`json:"Key"`)

**Pull request merged**